githooks/prepush.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+make pre-push
+
+# If the make command failed, prevent the push
+if [[ -z "$(git status -s)" ]]; then
```

Comment:
This returns true if the length of `git status -s` is 0 which would be the "good" state and you would not want to fail on that. Use `-n` which will be true if the length of git status is longer than 0

```suggestion
                    if [[ -n "$(git status -s)" ]]; then
```